documenting/automation/solution-pack/SP-Doc-Automation/contents.py

```diff
@@ -26,105 +31,248 @@ def __create_contents_file(self):
 
     def __get_contents_key_list(self):
         contents_key_list = []
-        for key, value in self.info_json_data['contents'].items():
+        for key, value in self.new_info_json_data['contents'].items():
             if value is not None:
                 contents_key_list.append(key)
         return contents_key_list
 
     def __create_contents_data(self, contents_file, contents_key_list):
 
-        # dashboards
+        # # dashboards
         dashboards_contents = ''
         if 'dashboards' in contents_key_list:
-            dashboards_contents = self.__create_dashboards_data()
-            logging.debug(
-                "Successfully added dashboard data in \"{0}\" file".format(CONTENTS_FILE_NAME))
+            dashboard_data = ''
+            new_dashboard_data = [dashboard["name"] for dashboard in self.new_info_json_data["contents"]["dashboards"]]
+            if self.is_new_version:
+                old_dashboard_data = [dashboard["name"] for dashboard in self.old_info_json_data["contents"]["dashboards"]]
+                diff_dashboard_data = list(set(new_dashboard_data) - set(old_dashboard_data))
+                dashboard_data = self.__create_name_and_description_data(diff_dashboard_data, True, dashboard_data)
+                dashboard_data = self.__create_name_and_description_data(old_dashboard_data, False, dashboard_data)
+            else:
+                dashboard_data = self.__create_name_and_description_data(new_dashboard_data, False, dashboard_data)
+            dashboards_contents = dashboards_data.substitute(dashboards_sub_data=dashboard_data)
+            logging.debug(f"Successfully fetched \"Dashboard\" data in {CONTENTS_FILE_NAME} file")
 
-        # picklist
+        # # picklist
```

Comment:
You should have created small func for each content with some generic handling

Reply:
For now made some logic generic, other will do it in new PR